### PR TITLE
RSE-873: Scheduled forecast GUI errors

### PR DESCRIPTION
**Pull request by avelasquezr**
**Is this a bugfix, or an enhancement? Please describe.**
This PR addresses a bug related to the Schedule Forecast page.

**Describe the problem**
When selecting more than two projects (but not all) on the Schedule Forecast page and clicking the "search" button, the loading icon remains stuck, and a warning icon appears. This issue stems from the API call to `http://rundeck/api/v/{params.project}/executions/running` where `params.project` is not correctly processed for multiple selected projects.

**Describe the solution you've implemented**
The solution involves two main changes:
1. **Interceptor Exclusion**: Modify the `ApiProjectSelectInterceptor` to include an exclusion for the `apiExecutionsRunningv14()` method, allowing it to bypass pre-validation checks that incorrectly handle multiple projects.
2. **API Method Enhancement**: The `apiExecutionsRunningv14()` method will be updated to:
   - Split the `params.project` string by commas to process each project individually.
   - Validate each project to ensure it exists, is active, and has read authorization.
   - Return appropriate error responses for invalid projects.

**Describe alternatives you've considered**
An alternative approach was to refactor the interceptor validations. However, this would have required significant changes across various API calls and potentially introduced new bugs. The chosen solution minimizes the impact on other components and focuses on addressing the specific issue.

**References**
- Related issue: `fixes #RSE-873`

### CODE REVIEW

File: functional-test/src/test/groovy/org/rundeck/util/container/BaseContainer.groovy [hunk 1]
```diff
@@ -310,6 +310,20 @@ abstract class BaseContainer extends Specification implements ClientProvider {
         }
     }
 
+    void disableScheduledAndDeleteProject(String projectName) {
```
Comment (ronaveva) [inline functional-test/src/test/groovy/org/rundeck/util/container/BaseContainer.groovy:313]: it would be better that the properties were received by parameter. This way  we could set any property instead of just this 2 properties

File: rundeckapp/src/test/groovy/rundeck/interceptors/ApiProjectSelectInterceptorSpec.groovy [hunk 2]
```diff
@@ -26,11 +26,21 @@ class ApiProjectSelectInterceptorSpec extends Specification implements Intercept
 
         where:
         project    | controller    | action                    | match
-        'xProject' | 'menu'        | 'apiExecutionsRunningv14' | true
         '*'        | 'menu'        | 'apiExecutionsRunningv14' | false
-        '***'      | 'menu'        | 'apiExecutionsRunningv14' | true
-        '*Project' | 'menu'        | 'apiExecutionsRunningv14' | true
```
Comment (ronaveva) [inline rundeckapp/src/test/groovy/rundeck/interceptors/ApiProjectSelectInterceptorSpec.groovy:32]: instead of delete, you could use match=false

File: rundeckapp/grails-app/controllers/rundeck/controllers/MenuController.groovy [hunk 3]
```diff
@@ -3578,10 +3578,27 @@ if executed in cluster mode.
...
+                def disabled = frameworkService.isFrameworkProjectDisabled(project)
+                if (disabled) {
+                    apiService.renderErrorFormat(response, [
+                            status: HttpServletResponse.SC_NOT_FOUND,
+                            code: 'api.error.project.disabled',
+                            args: [project]
+                    ])
+                    return true
+                }
+                def authorized = !apiAuthorizedForEventRead(project)
```
Comment (ronaveva) [inline rundeckapp/grails-app/controllers/rundeck/controllers/MenuController.groovy:3595]: you can use this directly in the if condition

File: functional-test/src/test/groovy/org/rundeck/util/gui/pages/BasePage.groovy [hunk 4]
```diff
@@ -126,11 +126,6 @@ abstract class BasePage {
                 .until(ExpectedConditions.elementToBeClickable(locator))
     }
 
-    WebElement waitForPresenceOfElementLocated(By locator) {
```
Comment (ronaveva) [inline functional-test/src/test/groovy/org/rundeck/util/gui/pages/BasePage.groovy:129]: why was this removed?

Comment (avelasquezr) [reply functional-test/src/test/groovy/org/rundeck/util/gui/pages/BasePage.groovy:129]: That method is not used in the project, neither in oss and pro.

File: functional-test/src/test/groovy/org/rundeck/util/container/BaseContainer.groovy [hunk 5]
```diff
@@ -303,17 +303,61 @@ abstract class BaseContainer extends Specification implements ClientProvider {
...
+     *
+     * @param projectName the name of the project to be disabled and deleted. Must not be null.
+     * @param body a map containing the configuration to be updated in the project before deletion.
+     *        Specifically, this map should include the necessary properties to disable
+     *        scheduled executions. The exact contents of the map will depend on the client API and
+     *        the project configuration.
+     * @throws RuntimeException if disabling scheduled executions or deleting the project fails.
+     *         The exception contains a detailed message obtained from the server's response.
+     */
+    void disableScheduledAndDeleteProject(String projectName, Map body) {
```
Comment (ronaveva) [inline functional-test/src/test/groovy/org/rundeck/util/container/BaseContainer.groovy:337]: method name and description does not match with the actual functionality, the method can be used to set any property value, not just disable schedule.
It would also be useful to have this separated into 2 methods, one to set the properties and delete on a separated mehtod

Comment (avelasquezr) [reply functional-test/src/test/groovy/org/rundeck/util/container/BaseContainer.groovy]: Okay, I have divided it into 2, this method changes the configuration of the project depending on the values you send it, and already exists the delete method.

File: rundeckapp/grails-app/controllers/rundeck/controllers/MenuController.groovy (hunk 3 above)
Comment (avelasquezr) [reply rundeckapp/grails-app/controllers/rundeck/controllers/MenuController.groovy]: Done

File: rundeckapp/src/test/groovy/rundeck/interceptors/ApiProjectSelectInterceptorSpec.groovy (hunk 2 above)
Comment (avelasquezr) [reply rundeckapp/src/test/groovy/rundeck/interceptors/ApiProjectSelectInterceptorSpec.groovy:32]: Done

File: rundeckapp/src/test/groovy/rundeck/controllers/MenuControllerSpec.groovy [hunk 6]
```diff
@@ -2607,6 +2686,56 @@ class MenuControllerSpec extends RundeckHibernateSpec implements ControllerUnitT
...
+            3 * isFrameworkProjectDisabled(_) >> false
+        }
+
+        controller.rundeckAuthContextProcessor=Mock(AppAuthContextProcessor){
+            3 * authorizeProjectResource(_, AuthorizationUtil.resourceType('event'),'read', _) >> true
+            0 * getAuthContextForSubject(_) >> test
+        }
+
+        params.project = 'project1,project2,project3'
+        request.api_version = 35
```
[the remark this answers is not in the document]
Comment (avelasquezr) [reply rundeckapp/src/test/groovy/rundeck/controllers/MenuControllerSpec.groovy]: Done, changed to 47.

File: functional-test/src/test/groovy/org/rundeck/util/container/BaseContainer.groovy (hunk 1 above)
Comment (avelasquezr) [reply functional-test/src/test/groovy/org/rundeck/util/container/BaseContainer.groovy]: Done

**Review by ronaveva — APPROVED** [review]
LGTM
